### DISCUSSION TITLE
Preferred strategy for Google?

### DISCUSSION THREAD

**Discussion by axhamre** [jaredhanson/passport#897] [Help] [answered]
There are 8 strategies for authenticating with Google. Why are there so many, and is there any preferred strategy?

**Comment by jaredhanson** [comment, marked as answer]
The preferred strategy today would be [passport-google-oidc](https://www.passportjs.org/packages/passport-google-oidc/).  There are many because over the past 10+ years Google has supported (and often deprecated) many different protocols (OpenID, OAuth 1, OAuth 2, OpenID Connect) and had various social efforts (Google+, etc.).

**Reply by axhamre** [reply]
Thank you! 🙏
